lib/helpers/android-bundle-validator-helper.ts

```diff
@@ -22,14 +24,10 @@ export class AndroidBundleValidatorHelper implements IAndroidBundleValidatorHelp
 			const androidRuntimeInfo = this.$projectDataService.getNSValue(projectData.projectDir, TNS_ANDROID_RUNTIME_NAME);
 			const androidRuntimeVersion = androidRuntimeInfo ? androidRuntimeInfo.version : "";
 
-			if (androidRuntimeVersion) {
-				const shouldSkipCheck = !semver.valid(androidRuntimeVersion) && !semver.validRange(androidRuntimeVersion);
-				if (!shouldSkipCheck) {
-					const isAndroidBundleSupported = semver.gte(semver.coerce(androidRuntimeVersion), semver.coerce(AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION));
-					if (!isAndroidBundleSupported) {
-						this.$errors.failWithoutHelp(util.format(AndroidBundleValidatorMessages.RUNTIME_VERSION_TOO_LOW, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION));
-					}
-				}
+			if (androidRuntimeVersion &&
```

Comment:
Cool! I really like it. But we can make it even better.
```
const shouldThrowError = androidRuntimeVersion && this.isValidVersion(androidRuntimeVersion) && this.isVersionLowerThan(androidRuntimeVersion, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION);
if (shouldThrowError) { ... }
```
So no need to provide `semver.gte` function as a param to another function.

Reply:
`isVersionLowerThan` is actually misleading, because it does not imply for `gte` comparison, but `gt` comparison. I prefer sending a function to be used to compare the values. This makes the helper much more versatile for future use.

Reply:
1. "divide and conquer" approach
```
isVersionLowerThan(..)
isVersionLowerOrEqualThan(..)
isVersionGreaterThan(...)
isVersionGreaterOrEqualThan(...)
```

If we decide to change the implementation or add additional logic to some of the above functions, the change will affect **ONLY** the specific function.  So we will have simplification + more control over the affected parts.

2.  
```
compareCoerceVersions(androidRuntimeVersion, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION, semver.gte)
compareCoerceVersions(androidRuntimeVersion, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION, semver.gt)
compareCoerceVersions(androidRuntimeVersion, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION, semver.lte)
compareCoerceVersions(androidRuntimeVersion, AndroidBundleValidatorHelper.MIN_RUNTIME_VERSION, semver.lt)
```

If we decide to change the implementation of `compareCoerceVersions` it will affect **ALL** the calls above. 
So we will wonder "how this will affect the case when the function is called for example with `semver.gt`" or Will it affect the case when passed function is `semver.lte`. So it will be harder to read + less control over the affected parts.

Reply:
Even if I create a wrapper for each comparison `semver` has, it will underneath use the same private method that wraps the same logic (getting the coerce of each version and invoking a function). So changing the private method will affect the same amount of places, but indirectly.

Reply:
@rosen-vladimirov @DimitarTachev  I would appreciate your input on the matter. If the majority considers its better to wrap semver methods I will do it.